pandas/tests/series/test_constructors.py

```diff
@@ -625,6 +625,32 @@ def test_constructor_dict(self):
         expected.iloc[1] = 1
         assert_series_equal(result, expected)
 
+    def test_constructor_dict_nan_key(self):
+        # GH 18480
+        d = {1: 'a', 2: 'b', np.nan: 'c'}
+        result = Series(d).sort_index()
+        expected = Series(['a', 'b', 'c'], index=[1, 2, np.nan])
+        assert_series_equal(result, expected)
+
+        # Different NaNs:
+        d = {1: 'a', 2: 'b', float('nan'): 'c', float('nan'): 'd'}
+        result = Series(d).sort_values()
+        expected = Series(['a', 'b', 'c', 'd'], index=[1, 2, np.nan, np.nan])
+        assert_series_equal(result, expected)
+
+        # NaN vs. None:
+        d = {1: 'a', 2: 'b', np.nan: 'c', None: 'd'}
+        result = Series(d).sort_values()
+        expected = Series(['a', 'b', 'c', 'd'], index=[1, 2, np.nan, None])
+        assert_series_equal(result, expected)
+
+        # MultiIndex:
```

Comment:
I would add another test that parametrizes over the missing values specifically (with a fixed structured for d), e.g. parametrize over ``None, np.nan, float('nan')``